components/brave_welcome_ui/actions/welcome_actions.ts

```diff
@@ -15,3 +16,26 @@ export const goToTabRequested = (url: string, target: string) => action(types.GO
 })
 
 export const closeTabRequested = () => action(types.CLOSE_TAB_REQUESTED)
+
+export const changeDefaultSearchProvider = (searchProvider: string) => action(types.CHANGE_DEFAULT_SEARCH_PROVIDER, searchProvider)
+
+export const getSearchEngineProvidersStarted = () => action(types.IMPORT_DEFAULT_SEARCH_PROVIDERS_STARTED)
+
+export const getSearchEngineProvidersSuccess = (searchProviders: Array<Welcome.SearchEngineEntry>) => action(types.IMPORT_DEFAULT_SEARCH_PROVIDERS_SUCCESS, searchProviders)
+
+export const getSearchEngineProvidersFailure = () => action(types.IMPORT_DEFAULT_SEARCH_PROVIDERS_FAILURE)
+
+export const getSearchEngineProviders = () => {
+  return (dispatch: Dispatch) => {
+    dispatch(getSearchEngineProvidersStarted())
+
+    // @ts-ignore
```

Comment:
I couldn't figure out how to type the `sendWithPromise` method 😅- if anyone knows a workaround please reach out

Reply:
@imptrx it should be defined in [brave://resources/js/cr.js](brave://resources/js/cr.js). from there:

```js
  /**
   * A variation of chrome.send, suitable for messages that expect a single
   * response from C++.
   * @param {string} methodName The name of the WebUI handler API.
   * @param {...*} var_args Variable number of arguments to be forwarded to the
   *     C++ call.
   * @return {!Promise}
   */
  function sendWithPromise(methodName, var_args) {
    const args = Array.prototype.slice.call(arguments, 1);
    const promiseResolver = new PromiseResolver();
    const id = methodName + '_' + createUid();
    chromeSendResolverMap[id] = promiseResolver;
    chrome.send(methodName, [id].concat(args));
    return promiseResolver.promise;
  }
```

also what you think of adding an abstraction for `sendWithPromise` instead of having it in here? maybe a file named `welcomeUtils` or similar. not sure if reducer itself have access to the `window` object.